pkg/providers/instancetype/suite_test.go

```diff
@@ -759,13 +759,68 @@ var _ = Describe("InstanceType Provider", func() {
 		})
 	})
 
+	Context("ImageProvider + Image Family", func() {
+		DescribeTable("should select the right image for a given instance type",
+			func(instanceType string, imageFamily string, expectedImageDefinition string, expectedGalleryURL string) {
+				nC := test.AKSNodeClass()
+				nC.Spec.ImageFamily = lo.ToPtr(imageFamily)
+				coretest.ReplaceRequirements(nodePool, v1.NodeSelectorRequirement{
+					Key:      v1.LabelInstanceTypeStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{instanceType},
+				})
+				nodePool.Spec.Template.Spec.NodeClassRef = &corev1beta1.NodeClassReference{Name: nC.Name}
```

Comment:
Can this just assign the Name?

Reply:
i don't think so it needs to be a NodeClassReference, correct me if i am wrong but I dont think we can

Reply:
I think it already has NodeClassReference, overriding just the Name should work, but not critical